mailpoet/lib/Doctrine/EventListeners/NewsletterListener.php

```diff
@@ -0,0 +1,50 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Doctrine\EventListeners;
+
+use MailPoet\Entities\NewsletterEntity;
+use MailPoet\Entities\WpPostEntity;
+use MailPoet\WP\Functions as WPFunctions;
+use MailPoetVendor\Doctrine\ORM\Event\LifecycleEventArgs;
+
+class NewsletterListener {
+  /** @var WPFunctions */
+  private $wp;
```

Comment:
I noticed you used a typed property in the class EntityManagerFactory.php for the new property, which is pretty nice 👍. Maybe we could also use typed properties in this class and its test. What do you think?